### PR TITLE
[CARBONDATA-1969] Support Java API to create table and write data

### PR DESCRIPTION
**Pull request by jackylk**
It is nice to have Java API to create carbon table and write CSV data into the table.
Application can use this API to write data and then query by SparkSQL

In this PR, a new module called store-sdk is added. No changes to existing module.

This PR is on top of #1749 

 - [X] Any interfaces changed?
 new API added
 - [X] Any backward compatibility impacted?
 No
 - [X] Document update required?
Yes
 - [X] Testing done
Testcase added
 - [X] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
NA

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2491/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1266/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2659/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2504/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1280/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2668/

File: store/sdk/src/main/java/org/apache/carbondata/store/TableBuilder.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    thriftTableInfo.getFact_table().getSchema_evolution().getSchema_evolution_history()
+        .add(schemaEvolutionEntry);
+    FileFactory.FileType fileType = FileFactory.getFileType(schemaMetadataPath);
+    if (!FileFactory.isFileExist(schemaMetadataPath, fileType)) {
+      FileFactory.mkdirs(schemaMetadataPath, fileType);
+    }
+    ThriftWriter thriftWriter = new ThriftWriter(schemaFilePath, false);
+    thriftWriter.open();
+    thriftWriter.write(thriftTableInfo);
+    thriftWriter.close();
```
Comment (mohammadshahidkhan) [inline store/sdk/src/main/java/org/apache/carbondata/store/TableBuilder.java]: Its not safe here to call close without finally block
Writer close can not be ensured if any IOException occurs at  thriftWriter.write(thriftTableInfo)

File: store/sdk/src/test/scala/org/apache/carbondata/store/TestCarbonFileWriter.scala [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      .addColumn(new StructField("height", DataTypes.DOUBLE), false)
+      .create
+
+    val table = carbon.createTable("t1", schema, tablePath)
+    val segment = table.newBatchSegment()
+
+    segment.open()
+    val writer = segment.newWriter()
+    (1 to 1000).foreach { _ => writer.writeRow(Array[String]("amy", "1", "2.3")) }
+    writer.close()
```
Comment (mohammadshahidkhan) [inline store/sdk/src/test/scala/org/apache/carbondata/store/TestCarbonFileWriter.scala]: Stream close can not be ensured here without finally.

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3292/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3297/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2060/

**Comment by jackylk** [comment]
merged with #1798

**Pull request closed without merging**